modmail/utils/extensions.py

```diff
@@ -0,0 +1,79 @@
+# original source:
+# https://github.com/python-discord/bot/blob/a8869b4d60512b173871c886321b261cbc4acca9/bot/utils/extensions.py
+# MIT License 2021 Python Discord
+import importlib
+import inspect
+import logging
+import pkgutil
+import typing as t
+
+from modmail import extensions
+from modmail.config import CONFIG
+from modmail.log import ModmailLogger
+from modmail.utils.cogs import BOT_MODES, BotModes, ExtMetadata
+
+log: ModmailLogger = logging.getLogger(__name__)
+
+EXT_METADATA = ExtMetadata
+
+
+EXTENSIONS: t.Dict[str, t.Tuple[bool, bool]] = dict()
+NO_UNLOAD: t.List[str] = list()
+
+
+def determine_bot_mode() -> int:
+    """
+    Figure out the bot mode from the configuration system.
+
+    The configuration system uses true/false values, so we need to turn them into an integer for bitwise.
+    """
+    bot_mode = 0
+    for mode in BotModes:
+        if getattr(CONFIG.dev.mode, str(mode).split(".")[-1].lower(), True):
```

Comment:
This needs a comment. I have no idea what this does. I apologize for being a hypocrite, but this deserves like.. a 3 line explaining comment for what it's doing.